test/core/unit/engine_http.test.js

```diff
@@ -112,26 +118,210 @@ test('HTTP virtual user', function (t) {
 
   function onStarted() {
-    t.assert(true, 'started event emitted');
+    t.ok(true, 'started event emitted');
   }
 });
 
+test('extendedMetrics', (t) => {
+  const histograms = new Set();
+  const additionalMetrics = [
+    'engine.http.dns',
+    'engine.http.tcp',
+    'engine.http.tls',
+    'engine.http.total'
+  ];
+  const target = nock('http://localhost:8888').get('/').reply(200, 'ok');
+
+  const script = {
+    config: {
+      target: 'http://localhost:8888',
+      http: { extendedMetrics: true }
+    },
+    scenarios: [
+      {
+        flow: [
+          {
+            get: {
+              url: '/'
+            }
+          }
+        ]
+      }
+    ]
+  };
+
+  const engine = new HttpEngine(script);
+  const ee = new EventEmitter();
+  const runScenario = engine.createScenario(script.scenarios[0], ee);
+
+  ee.on('histogram', (name) => {
+    histograms.add(name);
+  });
+
+  runScenario({ vars: {} }, function userDone(err) {
+    if (err) {
+      t.fail();
+    }
+
+    additionalMetrics.forEach((metric) => {
+      t.ok(
+        histograms.has(metric),
+        `it should track additional metric ${metric}`
+      );
+    });
+
+    t.ok(target.isDone(), 'Should have made a request to /');
+    t.end();
+  });
+});
+
+test('gzip - request headers', (t) => {
+  const target = nock('http://localhost:8888')
+    .post('/')
+    .reply(201, function () {
+      t.ok(
+        'accept-encoding' in this.req.headers,
+        'sets the accept-encoding header if gzip is true'
+      );
+
+      return 'ok';
+    });
+
+  const script = {
+    config: {
+      target: 'http://localhost:8888'
+    },
+    scenarios: [
+      {
+        flow: [
+          {
+            post: {
+              url: '/',
+              json: { foo: 'bar' },
+              gzip: true
+            }
+          }
+        ]
+      }
+    ]
+  };
+
+  const engine = new HttpEngine(script);
+  const ee = new EventEmitter();
+  const runScenario = engine.createScenario(script.scenarios[0], ee);
+
+  runScenario({ vars: {} }, function userDone(err) {
+    if (err) {
+      t.fail();
+    }
+
+    t.ok(target.isDone(), 'Should have made a request to /');
+    t.end();
+  });
+});
+
+test('gzip - compressed responses', (t) => {
+  const target = nock('http://localhost:8888')
+    .get('/')
+    .reply(201, zlib.gzipSync('ok'), { 'content-encoding': 'gzip' });
+
+  const script = {
+    config: {
+      target: 'http://localhost:8888'
+    },
+    scenarios: [
+      {
+        flow: [
+          {
+            get: {
+              url: '/',
+              gzip: true
+            }
+          }
+        ]
+      }
+    ]
+  };
+
+  const engine = new HttpEngine(script);
+  const ee = new EventEmitter();
+  const runScenario = engine.createScenario(script.scenarios[0], ee);
+
+  runScenario({ vars: {} }, function userDone(err, context) {
+    if (err) {
+      t.fail();
+    }
+
+    t.equal(context._successCount, 1, 'it should handle compressed responses');
```

Comment:
* would it make sense to roll the check from the previous test (whether `accept-encoding` header is set) into here to reduce the amount of test code?
* `_successCount` does not seem to be used anywhere else in the codebase (even though it seems to be set by all engines) - would suggest not relying on it here and getting rid of it
* we're not actually testing that the response is decompressed by the client as expected here? (even checking `_successCount` does not tell us that we have the decompressed response)